### ISSUE TITLE
Patch size and n_patches calculation issue when grid is specified

### ISSUE THREAD

**Issue by alabamagan** [jeonsworld/ViT-pytorch#56]
https://github.com/jeonsworld/ViT-pytorch/blob/460a162767de1722a014ed2261463dbbc01196b6/models/modeling.py#L132C1-L136C31

Here it seems the calculation for `n_patches` and `patch_size` are incorrect. 
I think if 16 x 16 patches were assumed, the grid size is already determined? Or am I mistaken about the purpose of the `grid` config?

**Comment by cai-wenbo** [comment]
I just read the code and had the same question. Then found this unsolved issue, I'm quite confirmed that this piece of code is of wrong logic.

**Comment by alabamagan** [comment]
> I just read the code and had the same question. Then found this unsolved issue, I'm quite confirmed that this piece of code is of wrong logic.

Yeah I end up rewriting my own version. I end up using implementation by another person: https://github.com/junyuchen245/ViT-V-Net_for_3D_Image_Registration_Pytorch
